### PR TITLE
docs: `ChanUpgradeInit` and governance gating

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #5270 

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 1]
```diff
@@ -11,6 +11,32 @@ slug: /ibc/channel-upgrades
 Learn how to upgrade existing IBC channels.
 :::
 
+## Initializing a Channel Upgrade
+
+A channel upgrade is initialised by submitting the `ChanUpgradeInit` message, which can be submitted by relayer or the chain itself upon governance authorization. This message should specify an appropriate timeout window for the upgrade, in consideration of the governance method chosen (more in the section below).
```
Comment (crodriguezvega) [inline docs/docs/01-ibc/06-channel-upgrades.md:16]:
```suggestion
A channel upgrade is initialised by submitting the `ChanUpgradeInit` message, which can only be submitted by the chain itself upon governance authorization. This message should specify an appropriate timeout window for the upgrade, in consideration of the governance method chosen (more in the section below).
```

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 2]
```diff
@@ -11,6 +11,32 @@ slug: /ibc/channel-upgrades
...
+
+1. **A and B pre-approves certain upgrade types** : 
+
+    A technical committee/DAO on A, elected by chain governance, initiates a governance proposal using the groups module stating their chain pre approves all channel upgrade types of the transfer module from v1 to v2. If this proposal passes, and if chain B has also pre-approved upgrades of this type (transfer v1 to v2), the channel upgrade executes in a permissioned and pre-approved manner. In this case, any relayer can submit the `MsgChanUpgradeInit` and `MsgChanUpgradeTry`.
+
+2. **Governance-gated upgrades on A and pre-approved upgrade type on B**: 
+    
+    Chain A passes a governance proposal to approve an upgrade and executes `ChanUpgradeInit`. Chain B has pre-approved all upgrades of this type (initiated on A) and so any relayer can submit `MsgChanUpgradeTry` on B.
+
+3. **Governance-gated upgrades on A and B**:
```
[the remark this answers is not in the document]
Comment (charleenfei) [reply docs/docs/01-ibc/06-channel-upgrades.md:34]: ah, i didnt check that it's not going to be in the first release. will update!

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 3]
```diff
@@ -11,6 +11,20 @@ slug: /ibc/channel-upgrades
...
+
+A channel upgrade is initialised by submitting the `ChanUpgradeInit` message, which can be submitted only by the chain itself upon governance authorization. This message should specify an appropriate timeout window for the upgrade.
+
+As part of the handling of the `ChanUpgradeInit` message, the application's callbacks `OnChanUpgradeInit` will be triggered as well.
+
+After this message is handled successfully, the channel's upgrade sequence will be incremented. This upgrade sequence will serve as a nonce for the upgrade process to avoid error receipt proof collisions.
+
+### Governance Gating on `ChanUpgradeInit`
+
+A message signer who has been designated as the `authority` of the `IBCKeeper` sending the `MsgChanUpgradeInit`. If this proposal passes, the counterparty channels will upgrade by default.
```
Comment (chatton) [inline docs/docs/01-ibc/06-channel-upgrades.md:24]: There is a single channel version being upgraded per `MsgChanUpgradeInit` right?

```suggestion
A message signer who has been designated as the `authority` of the `IBCKeeper` sending the `MsgChanUpgradeInit`. If this proposal passes, the counterparty's channel will upgrade by default.
```

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 4]
```diff
@@ -11,6 +11,20 @@ slug: /ibc/channel-upgrades
...
+
+If chains want to initiate the upgrade of many channels, they will need to submit a governance proposal with multiple `MsgChanUpgradeInit`, one for each channel they would like to upgrade, again with message signer as the designated `authority` of the `IBCKeeper`
```
Comment (chatton) [inline docs/docs/01-ibc/06-channel-upgrades.md:26]:
```suggestion
If chains want to initiate the upgrade of many channels, they will need to submit a governance proposal with multiple `MsgChanUpgradeInit` messages, one for each channel they would like to upgrade, again with message signer as the designated `authority` of the `IBCKeeper`
```

**Review by chatton — APPROVED** [review]
LGTM, just a few small nits

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 5]
```diff
@@ -11,6 +11,20 @@ slug: /ibc/channel-upgrades
...
 
+## Initializing a Channel Upgrade
+
+A channel upgrade is initialised by submitting the `ChanUpgradeInit` message, which can be submitted only by the chain itself upon governance authorization. This message should specify an appropriate timeout window for the upgrade.
+
+As part of the handling of the `ChanUpgradeInit` message, the application's callbacks `OnChanUpgradeInit` will be triggered as well.
+
+After this message is handled successfully, the channel's upgrade sequence will be incremented. This upgrade sequence will serve as a nonce for the upgrade process to avoid error receipt proof collisions.
+
+### Governance Gating on `ChanUpgradeInit`
```
Comment (colin-axner) [inline docs/docs/01-ibc/06-channel-upgrades.md:22]:
```suggestion
### Governance gating on `ChanUpgradeInit`
```
I believe titles should only have the first word capitalized

File: docs/docs/01-ibc/06-channel-upgrades.md (hunk 3 above)
Comment (colin-axner) [inline docs/docs/01-ibc/06-channel-upgrades.md:24]:
```suggestion
The message signer for `MsgChanUpgradeInit must be the address which has been designated as the `authority` of the `IBCKeeper` . If this proposal passes, the counterparty channels will upgrade by default.
```

**Review by colin-axner — APPROVED** [review]
LGTM, see suggestions

**Pull request merged**